### ISSUE TITLE
Timeout error connecting to CosmosDB Mongo API

### ISSUE THREAD

**Issue by dansbits** [elbywan/cryomongo#14]
I'm getting a timeout error when I try to connect to an instance of an Azure CosmosDB. I'm not sure where to start trying to debug this but curious if you have any thoughts. My app is very simple. This is running in a simple Kemal app. The full app is below.

```
require "kemal"
require "cryomongo"

client = Mongo::Client.new ENV["MONGODB_URL"]

get "/featured_posts" do |env|
  database = client["blog"]
  postsCollection = database["posts"]
  projection = { title: 1, text: 1, createdAt: 1, updatedAt: 1 }
  posts = postsCollection.find(
    { title: { "$exists": true }},
    projection: projection,
    limit: 20,
    sort: { createdAt: 1 }
  ).to_a
  env.response.content_type = "application/json"
  posts.to_json
end

port = ENV["PORT"] || "80"
Kemal.run port.to_i
```

I see the following error when I try to load the page.
```
Timeout (00:00:30) reached while trying to select a suitable server with read preference primary.
```

Here is my shards.yml
```
name: Blog
version: 0.0.1

targets:
  app:
    main: src/app.cr
dependencies:
  kemal:
    github: kemalcr/kemal
  cryomongo:
    github: elbywan/cryomongo

crystal: 1.3.2

license: MIT

```

**Comment by elbywan** [comment]
Hey @dansbits,

So I've never tried connecting with cryomongo to a cosmosdb instance before, and I had to setup a free account to try it out.

I spotted a couple of small specific issues that are now fixed with the `v0.3.10` release.

> Timeout (00:00:30) reached while trying to select a suitable server with read preference primary.

I did not run into this error though, are you using the right connection string with the proper mongo options (ssl=true)?  Using the [provided one](https://docs.microsoft.com/en-us/azure/cosmos-db/mongodb/connect-mongodb-account#get-the-mongodb-connection-string-to-customize) works fine on my end.
